### ISSUE TITLE
Dagger 2.31.1 depends on RC versions of AndroidX modules

### ISSUE THREAD

**Issue by francescocervone** [google/dagger#2300]
I don't know if this is intentional, but I believe that majority of client projects prefer to depend on stable releases.

[Diff](https://github.com/google/dagger/compare/dagger-2.31...dagger-2.31.1#diff-5493ff8e9397811510e780de47c57abb70137f1afe85d1519130dc3679d60ce5)

**Comment by shekibobo** [comment]
This has been throwing me off all morning. Our fragment animations apparently don't exist in `fragment:1.3.0-alphaX`.

**Comment by danysantiago** [comment]
This is intentional, the plan is to declare these dependencies on the stable androidx versions before Hilt is moved to beta and stable. Note that these androidx RC releases are API locked, meaning whichever APIs are no longer available will also not be available in the stable release. See https://developer.android.com/jetpack/androidx/versions for clarification of androidx versions.

**Comment by francescocervone** [comment]
Thanks @danysantiago. This does make sense if you talk about Hilt which an alpha dependency, but I'm pretty sure also `dagger-android` users are affected by this change. Am I wrong?

**Comment by danysantiago** [comment]
Ah, Yes, `dagger-android` is also affected here and that was not intended. We are looking into resolving this issue.

**Comment by sczerwinski** [comment]
The biggest problem these RC versions of dependencies cause is with Kotlin multiplatform projects:
```
* What went wrong:
   > Could not resolve androidx.activity:activity:1.2.0-rc01.
     Required by:
         project :hilt:processor > com.google.dagger:hilt-android:2.31.1-alpha
      > No matching variant of androidx.activity:activity:1.2.0-rc01 was found. The consumer was configured to find an API of a library compatible with Java 8, preferably in the form of class files, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' but:
          - Variant 'releaseApiPublication' capability androidx.activity:activity:1.2.0-rc01 declares an API of a library, and its dependencies declared externally:
              - Incompatible because this component declares a component, with the library elements 'aar' and the consumer needed a component, preferably in the form of class files
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about org.jetbrains.kotlin.platform.type (required 'jvm')
          - Variant 'releaseRuntimePublication' capability androidx.activity:activity:1.2.0-rc01 declares a runtime of a library, and its dependencies declared externally:
              - Incompatible because this component declares a component, with the library elements 'aar' and the consumer needed a component, preferably in the form of class files
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about org.jetbrains.kotlin.platform.type (required 'jvm')
          - Variant 'sourcesElements' capability androidx.activity:activity:1.2.0-rc01 declares a runtime of a component, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them preferably in the form of class files)
                  - Doesn't say anything about org.jetbrains.kotlin.platform.type (required 'jvm')
   > Could not resolve androidx.fragment:fragment:1.3.0-rc01.
     Required by:
         project :hilt:processor > com.google.dagger:hilt-android:2.31.1-alpha
      > No matching variant of androidx.fragment:fragment:1.3.0-rc01 was found. The consumer was configured to find an API of a library compatible with Java 8, preferably in the form of class files, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' but:
          - Variant 'releaseApiPublication' capability androidx.fragment:fragment:1.3.0-rc01 declares an API of a library, and its dependencies declared externally:
              - Incompatible because this component declares a component, with the library elements 'aar' and the consumer needed a component, preferably in the form of class files
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them preferably in the form of class files)
                  - Doesn't say anything about org.jetbrains.kotlin.platform.type (required 'jvm')
   > Could not resolve androidx.savedstate:savedstate:1.1.0-rc01.
     Required by:
         project :hilt:processor > com.google.dagger:hilt-android:2.31.1-alpha
      > No matching variant of androidx.savedstate:savedstate:1.1.0-rc01 was found. The consumer was configured to find an API of a library compatible with Java 8, preferably in the form of class files, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' but:
          - Variant 'releaseApiPublication' capability androidx.savedstate:savedstate:1.1.0-rc01 declares an API of a library, and its dependencies declared externally:
              - Incompatible because this component declares a component, with the library elements 'aar' and the consumer needed a component, preferably in the form of class files
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about org.jetbrains.kotlin.platform.type (required 'jvm')
          - Variant 'releaseRuntimePublication' capability androidx.savedstate:savedstate:1.1.0-rc01 declares a runtime of a library, and its dependencies declared externally:
              - Incompatible because this component declares a component, with the library elements 'aar' and the consumer needed a component, preferably in the form of class files
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about org.jetbrains.kotlin.platform.type (required 'jvm')
          - Variant 'sourcesElements' capability androidx.savedstate:savedstate:1.1.0-rc01 declares a runtime of a component, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them preferably in the form of class files)
                  - Doesn't say anything about org.jetbrains.kotlin.platform.type (required 'jvm')
```

**Comment by Chang-Eric** [comment]
Sorry for the problems here, this change wasn't supposed to make it into the minor release between 2.31 and 2.31.1. This has been reverted and that has been released in 2.31.2 now.
